File: src/cse/ele/EleValue.java
package cse.ele;

import Interpreter.Node;

import java.util.Objects;

/**
 * EleValue which will store all elements except tuples.
 */
public class EleValue extends EleValueOrTuple {
    private final String value;

    /**
     * Create element from labels
     */
    public EleValue(String label) {
        super(label);
        this.value = null;
    }

    /**
     * Create element from label and value
     */
    public EleValue(String label, String value) {
        super(label);
        this.value = value;
    }

    /**
     * Create element from node
     */
    public EleValue(Node node) {
        super(node.getLabel());
        this.value = node.getValue();
    }

    public String getValue() {
        return value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EleValue that = (EleValue) o;
        return Objects.equals(value, that.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(value);
    }

    @Override
    public String toString() {
        if (value == null) return getLabel();
        return String.format("%s(%s)", getLabel(), getValue());
    }
}

File: reCreate/main/CSE/elements/EleValue.py
from .EleValueOrTuple import EleValueOrTuple
from INTERPRETER.Node import Node
 
class EleValue(EleValueOrTuple):
    def __init__(self, type, value=None):
        if isinstance(type, str):
             super().__init__(type)
             self.value = value
        if isinstance(type, Node):
            super().__init__(type.getLabel())
            self.value = type.getValue()

    def getValue(self):
        return self.value

    def equals(self, o):
        if self is o:
            return True
        if o is None or self.__class__ != o.__class__:
            return False
        return self.value == o.value

    def hashCode(self):
        return hash(self.value)

    def toString(self):
        if self.value is None:
            return self.getLabel()
        return f"{self.getLabel()}({self.getValue()})"
